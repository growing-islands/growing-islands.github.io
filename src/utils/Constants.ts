type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
