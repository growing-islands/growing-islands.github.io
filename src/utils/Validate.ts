/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
