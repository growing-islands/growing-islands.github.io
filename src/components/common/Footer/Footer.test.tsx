import { render } from '~tests/libs/utils'
import { Footer } from '.'

describe('components/common/Footer', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<Footer />)

    expect(asFragment()).toMatchSnapshot()
  })
})
