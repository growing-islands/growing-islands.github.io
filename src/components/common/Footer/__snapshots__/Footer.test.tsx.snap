// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Footer Snapshot 1`] = `
<DocumentFragment>
  <footer
    class=""
    data-testid="component"
  >
    <div>
      <small
        translate="no"
      >
        Â©Yume Lab Inc.
      </small>
      <nav>
        <ul
          class=""
        >
          <li>
            <a
              href="https://twitter.com/yumelab1214"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-label="Twitter"
                class=""
                viewBox="0 0 512 415.84"
              >
                <path
                  d="M459.37 103.64c.33 4.54.33 9.09.33 13.64 0 138.72-105.59 298.56-298.56 298.56A296.51 296.51 0 0 1 0 368.73a215.83 215.83 0 0 0 25.34 1.3 210.15 210.15 0 0 0 130.27-44.83 105.12 105.12 0 0 1-98.11-72.77 132.94 132.94 0 0 0 19.82 1.62 111.09 111.09 0 0 0 27.61-3.57 105 105 0 0 1-84.14-103v-1.29a105.67 105.67 0 0 0 47.43 13.31A105.07 105.07 0 0 1 35.74 19.17 298.28 298.28 0 0 0 252.1 128.98a118.78 118.78 0 0 1-2.6-24 105 105 0 0 1 181.6-71.8 206.58 206.58 0 0 0 66.6-25.34 104.61 104.61 0 0 1-46.13 57.83A210.61 210.61 0 0 0 512 49.38a225.62 225.62 0 0 1-52.63 54.26Z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://www.facebook.com/yumelab.co.jp/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-label="Facebook"
                class=""
                viewBox="0 0 448 448"
              >
                <path
                  d="M400 0H48A48 48 0 0 0 0 48v352a48 48 0 0 0 48 48h137.25V295.69h-63V224h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V224h68.78l-11 71.69h-57.78V448H400a48 48 0 0 0 48-48V48a48 48 0 0 0-48-48Z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </footer>
</DocumentFragment>
`;
